### PR TITLE
Add golangci-lint and fixup flagged issues

### PR DESCRIPTION
**Pull request by stuartleeks**
Addresses #111

### CODE REVIEW

File: controllers/secretscope_controller_databricks.go [hunk 1]
```diff
@@ -180,15 +178,6 @@ func (r *SecretScopeReconciler) submit(instance *databricksv1alpha1.SecretScope)
 	return r.Update(context.Background(), instance)
 }
 
-func (r *SecretScopeReconciler) update(instance *databricksv1alpha1.SecretScope) error {
```
Comment (Azadehkhojandi) [inline controllers/secretscope_controller_databricks.go]: Why the update function is being deleted?

File: Makefile [hunk 2]
```diff
@@ -78,12 +78,12 @@ manifests: controller-gen
...
 fmt:
-	go fmt ./...
-
-# Run go vet against code
-vet:
-	go vet ./...
+	find . -name '*.go' | grep -v vendor | xargs gofmt -s -w
 
+# Run linting
+checks:
```
Comment (Azadehkhojandi) [inline Makefile]: Could you please rename checks to lint?

File: Makefile [hunk 3]
```diff
@@ -78,12 +78,12 @@ manifests: controller-gen
 
 # Run go fmt against code
 fmt:
-	go fmt ./...
-
-# Run go vet against code
-vet:
```
Comment (Azadehkhojandi) [inline Makefile]: Is it possible to keep vet?

File: Makefile [hunk 4]
```diff
@@ -78,12 +78,12 @@ manifests: controller-gen
 
 # Run go fmt against code
 fmt:
-	go fmt ./...
```
Comment (Azadehkhojandi) [inline Makefile]: what was the reason of changing `go fmt ./...`  to `find . -name '*.go' | grep -v vendor | xargs gofmt -s -w`?

File: Makefile [hunk 5]
```diff
@@ -33,11 +33,11 @@ test-existing: generate fmt vet manifests
 	rm -f cover.out cover.out.tmp cover.json
 
 # Build manager binary
-manager: generate fmt vet
+manager: generate fmt checks
 	go build -o bin/manager main.go
 
 # Run against the configured Kubernetes cluster in ~/.kube/config
-run: generate fmt vet
+run: generate fmt checks
```
Comment (Azadehkhojandi) [inline Makefile]: Is it possible to have `generate fmt vet lint` instead of `generate fmt checks`?

File: Makefile [hunk 6]
```diff
@@ -20,7 +20,7 @@ test: generate fmt vet manifests
 	rm -f cover.out cover.out.tmp cover.json
 
 # Run tests with existing cluster
-test-existing: generate fmt vet manifests
+test-existing: generate fmt checks manifests
```
Comment (Azadehkhojandi) [inline Makefile]: Is it possible to have ` generate fmt vet manifests lint` instead of ` generate fmt checks manifests`?

File: Makefile [hunk 7]
```diff
@@ -7,7 +7,7 @@ CRD_OPTIONS ?= "crd:trivialVersions=true"
 all: manager
 
 # Run tests
-test: generate fmt vet manifests
+test: generate fmt checks manifests
```
Comment (Azadehkhojandi) [inline Makefile]: Is it possible to have ` generate fmt vet manifests lint` instead of ` generate fmt checks manifests`?

File: .golangci.toml [hunk 8]
```diff
@@ -0,0 +1,42 @@
+[run]
```
Comment (Azadehkhojandi) [inline .golangci.toml]: Is it possible to have the config in yaml format?

File: controllers/secretscope_controller_databricks.go (hunk 1 above)
Comment (stuartleeks) [reply controllers/secretscope_controller_databricks.go]: The linter flagged that it wasn't used anywhere :-)

File: Makefile (hunk 4 above)
Comment (stuartleeks) [reply Makefile]: the golangci-lint checks that the code has been simplified and I reused the fmt command from other projects to apply the simplification and to ignore vendored packages (avoids linter failures on those if vendoring is ever used)

File: Makefile (hunk 3 above)
Comment (stuartleeks) [reply Makefile]: golangci-lint includes govet as part of its checks (and is enabled in the config I added) so I didn't think that added much value?

File: .golangci.toml (hunk 8 above)
Comment (stuartleeks) [reply .golangci.toml]: done :-)

File: Makefile (hunk 2 above)
Comment (stuartleeks) [reply Makefile]: Done :-)

File: Makefile (hunk 5 above)
Comment (stuartleeks) [reply Makefile]: As mentioned above, golangci-lint includes go vet. Let me know if you still want go vet added back

File: Makefile (hunk 6 above)
Comment (stuartleeks) [reply Makefile]: As mentioned above, golangci-lint includes go vet. Let me know if you still want go vet added back

File: Makefile (hunk 3 above)
Comment (stuartleeks) [reply Makefile]: Re-added as per discussion

File: Makefile (hunk 5 above)
Comment (stuartleeks) [reply Makefile]: Re-added as per discussion

File: Makefile (hunk 7 above)
Comment (stuartleeks) [reply Makefile]: Re-added as per discussion

File: Makefile (hunk 6 above)
Comment (stuartleeks) [reply Makefile]: Re-added as per discussion

**Pull request merged**